arkane/input.py

```diff
@@ -659,24 +661,49 @@ def loadInputFile(path):
-            job.modelChemistry = model_chemistry.lower()
+            job.modelChemistry = sp_level
             job.frequencyScaleFactor = frequency_scale_factor
             job.includeHinderedRotors = use_hindered_rotors
             job.applyAtomEnergyCorrections = use_atom_corrections
             job.applyBondEnergyCorrections = use_bond_corrections
             job.atomEnergies = atom_energies
         if isinstance(job, ThermoJob):
             job.arkane_species.author = author
-            job.arkane_species.level_of_theory = level_of_theory
-            if '//' in level_of_theory:
-                level_of_theory_energy = level_of_theory.split('//')[0]
-                if level_of_theory_energy != model_chemistry:
-                    # Only log the model chemistry if it isn't identical to the first part of level_of_theory
-                    job.arkane_species.model_chemistry = model_chemistry
+            job.arkane_species.level_of_theory = model_chemistry
             job.arkane_species.frequency_scale_factor = frequency_scale_factor
             job.arkane_species.use_hindered_rotors = use_hindered_rotors
             job.arkane_species.use_bond_corrections = use_bond_corrections
             if atom_energies is not None:
                 job.arkane_species.atom_energies = atom_energies
 
     return jobList, reactionDict, speciesDict, transitionStateDict, networkDict
+
+
+def process_model_chemistry(model_chemistry):
+    """Process the model chemistry string representation
+
+    Args:
+        model_chemistry (str, unicode): A representation of the model chemistry in an sp//freq format
+                                        e.g., 'CCSD(T)-F12a/aug-cc-pVTZ//B3LYP/6-311++G(3df,3pd)',
+                                        or a composite method, e.g. 'CBS-QB3'.
+
+    Returns:
+        str, unicode: The single point energy level of theory
+        str, unicode: The frequency level of theory
+    """
+    if model_chemistry.count('//') > 1:
+        raise InputError('The model chemistry seems wrong. It should either be a composite method (like CBS-QB3) '
+                         'or of the form sp//geometry, e.g., CCSD(T)-F12a/aug-cc-pVTZ//B3LYP/6-311++G(3df,3pd), '
+                         'and should not contain more than one appearance of "//".\n'
+                         'Got: {0}'.format(model_chemistry))
+    elif '//' in model_chemistry:
+        # assume this is an sp//freq format, split
+        sp_level, freq_level = model_chemistry.split('//')
+    else:
+        # assume the sp and freq levels are the same, assign the model chemistry to both
+        # (this could also be a composite method, and we'll expect the same behavior)
+        sp_level = freq_level = model_chemistry
+        if model_chemistry.startswith('cbs-qb3'):
+            # hard code for CBS-QB3-Paraskevas which has the same frequency scaling factor as CBS-QB3
+            freq_level = 'cbs-qb3'
```

Comment:
Sorry about seeing this after merging, but this hard-coding is slightly redundant since there's an entry for `cbs-qb3-paraskevas` in `assign_frequency_scale_factor` (https://github.com/ReactionMechanismGenerator/RMG-Py/blob/master/arkane/statmech.py#L1224).

I think we could remove one or the other.

Reply:
Sorry, didn't check the `assign_frequency_scale_factor` function in detail.

Reply:
You're absolutely right, I haven't noticed the redundancy either.